sssom/util.py

```diff
@@ -277,7 +277,7 @@ def filter_redundant_rows(
     # create a 'sort' method and then replce the following line by sort()
     df = sort_sssom(df)
     # df[CONFIDENCE] = df[CONFIDENCE].apply(lambda x: x + random.random() / 10000)
-    df, nan_df = assign_default_confidence(df)
+    df, nan_df, confidence_column_created = assign_default_confidence(df)
```

Comment:
Could you keep track of wether there was a confidence column in the context of filter redundant rows and avoid creating the confidence column created var?

Reply:
I could drop the `confidence` column at the end `if` all rows are `0.0` for that column but that's making an assumption that it happens only when we add a `confidence` column.

Reply:
So this is not an option?

```
    confidence_in_original = CONFIDENCE in df.columns
    df, nan_df = assign_default_confidence(df)
    if not confidence_in_original:
         return_df = return_df.drop(columns=[CONFIDENCE], axis=1)
```

Maybe I am misunderstanding something

Reply:
Aah .... I thought you didn't want a new variable in general. We can do this.

Reply:
Done